dbapi/src/main/java/com/griddynamics/jagger/dbapi/DatabaseServiceImpl.java

```diff
@@ -1260,6 +1265,64 @@ public SessionInfoProvider getSessionInfoService() {
     }
 
     @Override
+    public DecisionPerSessionDto getDecisionPerSession(String sessionId) {
+        DecisionPerSessionEntity sessionDecision = (DecisionPerSessionEntity)
+                entityManager.createQuery("select dps from DecisionPerSessionEntity as dps where dps.sessionId = :sessionId")
+                        .setParameter("sessionId", sessionId).getSingleResult();
+
+        List<DecisionPerTaskEntity> taskDecisions = (List<DecisionPerTaskEntity>)
+                entityManager.createQuery("select dpt from DecisionPerTaskEntity as dpt where dpt.taskData.sessionId = :sessionId")
+                        .setParameter("sessionId", sessionId).getResultList();
+
+        Set<Long> taskIds = taskDecisions.stream().map(decision -> decision.getTaskData().getId()).collect(Collectors.toSet());
+        Map<Long, Set<Long>> testIdsByTestGroupIds = getTestGroupIdsByTestIds(taskIds);
+
+        List<DecisionPerTestGroupDto> decisionPerTestGroupDtos = new ArrayList<>();
+        List<DecisionPerTestDto> decisionPerTestDtos = new ArrayList<>();
+        for (DecisionPerTaskEntity taskDecision : taskDecisions) {
+            // get metric decisions for task
+            List<DecisionPerMetricEntity> metricDecisions = (List<DecisionPerMetricEntity>) entityManager.createQuery(
+                    "select dpm from DecisionPerMetricEntity as dpm where dpm.metricDescriptionEntity.taskData.id = :taskId")
+                    .setParameter("taskId", taskDecision.getTaskData().getId()).getResultList();
+
+            // create DecisionPerTestDto and add it to list
+            DecisionPerTestDto decisionPerTestDto = new DecisionPerTestDto(taskDecision);
+            decisionPerTestDto.setMetricDecisions(metricDecisions.stream().map(DecisionPerMetricDto::new).collect(toList()));
+            decisionPerTestDtos.add(decisionPerTestDto);
```

Comment:
do I understand it right, decisions per test group are also duplicated in decisions per test array?